From 516af5b304d38efb447261499beb4bbea695402c Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Sun, 21 Oct 2018 09:00:54 -0600
Subject: [PATCH 16/18] msm8974-common: sepolicy: Resolve init denials

* avc: denied { add } for interface=android.hidl.base::IBase pid=337
  scontext=u:r:init:s0 tcontext=u:object_r:hidl_base_hwservice:s0
  tclass=hwservice_manager permissive=1
* avc: denied { add } for interface=android.hardware.sensors::ISensors
  pid=337 scontext=u:r:init:s0
  tcontext=u:object_r:hal_sensors_hwservice:s0 tclass=hwservice_manager
  permissive=1
* avc: denied { find } for interface=android.hardware.sensors::ISensors
  pid=337 scontext=u:r:init:s0
  tcontext=u:object_r:hal_sensors_hwservice:s0 tclass=hwservice_manager
  permissive=1
* avc: denied { write } for name="enable_adaptive_lmk" dev="sysfs"
  ino=6724 scontext=u:r:init:s0
  tcontext=u:object_r:sysfs_lowmemorykiller:s0 tclass=file permissive=1
* avc: denied { open } for name="enable_adaptive_lmk" dev="sysfs"
  ino=6724 scontext=u:r:init:s0
  tcontext=u:object_r:sysfs_lowmemorykiller:s0 tclass=file permissive=1
* avc: denied { read write } for name="accel_poll_delay" dev="sysfs"
  ino=27827 scontext=u:r:init:s0 tcontext=u:object_r:sysfs:s0
  tclass=file permissive=1
* avc: denied { setattr } for name="firmware_path" dev="sysfs"
  ino=6423 scontext=u:r:init:s0
  tcontext=u:object_r:sysfs_wifi_writeable:s0 tclass=file
  permissive=1
* avc: denied { write } for name="l2" dev="sysfs" ino=29063
  scontext=u:r:init:s0 tcontext=u:object_r:sysfs_msm_power:s0
  tclass=file permissive=1
* avc: denied { open } for name="l2" dev="sysfs" ino=29063
  scontext=u:r:init:s0 tcontext=u:object_r:sysfs_msm_power:s0
  tclass=file permissive=1
* avc: denied { write } for name="enabled" dev="sysfs" ino=29716
  scontext=u:r:init:s0 tcontext=u:object_r:sysfs_thermal:s0
  tclass=file permissive=1
* avc: denied { write } for name="online" dev="sysfs" ino=5871
  scontext=u:r:init:s0 tcontext=u:object_r:sysfs_devices_system_cpu:s0
  tclass=file permissive=1
* avc: denied { write } for name="boost_ms" dev="sysfs" ino=6652
  scontext=u:r:init:s0 tcontext=u:object_r:sysfs_cpu_boost:s0
  tclass=file permissive=1
* avc: denied { open } for name="boost_ms" dev="sysfs" ino=6652
  scontext=u:r:init:s0 tcontext=u:object_r:sysfs_cpu_boost:s0
  tclass=file permissive=1
* avc: denied { setattr } for name="file" dev="sysfs" ino=23591
  scontext=u:r:init:s0 tcontext=u:object_r:sysfs:s0 tclass=file
  permissive=1
* avc: denied { setattr } for name="booster" dev="sysfs" ino=23129
  scontext=u:r:init:s0 tcontext=u:object_r:sysfs:s0 tclass=file
  permissive=1
* avc: denied { ioctl } for path="/dev/iio:device4" dev="tmpfs"
  ino=7277 ioctlcmd=4540 scontext=u:r:init:s0
  tcontext=u:object_r:iio_device:s0 tclass=chr_file permissive=1
* avc: denied { setattr } for name="min_pwrlevel" dev="sysfs"
  ino=19546 scontext=u:r:init:s0 tcontext=u:object_r:sysfs_kgsl:s0
  tclass=file permissive=0
* avc: denied { setattr } for name="enabled" dev="sysfs" ino=23417
  scontext=u:r:init:s0 tcontext=u:object_r:sysfs_hal_pwr:s0
  tclass=file permissive=1
* avc: denied { setattr } for name="rear_camfw" dev="sysfs" ino=24404
  scontext=u:r:init:s0 tcontext=u:object_r:sysfs_camera:s0
  tclass=file permissive=1
* avc: denied { check_context } for scontext=u:r:init:s0
  tcontext=u:object_r:kernel:s0 tclass=security permissive=0

Change-Id: Id7f78abedea2209f84527b1b83259574d06a0900
---
 sepolicy/common/file.te       |  2 ++
 sepolicy/common/file_contexts |  4 ++++
 sepolicy/common/init.te       | 24 ++++++++++++++++++++++--
 3 files changed, 28 insertions(+), 2 deletions(-)

diff --git a/sepolicy/common/file.te b/sepolicy/common/file.te
index 52a6e4b..b37830a 100644
--- a/sepolicy/common/file.te
+++ b/sepolicy/common/file.te
@@ -15,6 +15,8 @@ type sysfs_sec_switch, fs_type, sysfs_type;
 type sysfs_sec_thermistor, fs_type, sysfs_type;
 type sysfs_sec_touchkey, fs_type, sysfs_type;
 type sysfs_sec_tsp, fs_type, sysfs_type;
+type sysfs_usb_otg, fs_type, sysfs_type;
+type sysfs_usb_storage_gadget, fs_type, sysfs_type;
 type sysfs_wifi_writeable, fs_type, sysfs_type;
 
 type bt_fw_file, file_type;
diff --git a/sepolicy/common/file_contexts b/sepolicy/common/file_contexts
index f2f9bda..90b896b 100644
--- a/sepolicy/common/file_contexts
+++ b/sepolicy/common/file_contexts
@@ -89,3 +89,7 @@
 
 # sysfs - sensors
 /sys/devices/virtual/sensors(/.*)?                      u:object_r:sysfs_sensors:s0
+
+# sysfs - usb
+/sys/devices/msm_dwc3/[a-f0-9]+\.dwc3/gadget/lun[0-9]+(/.*)? u:object_r:sysfs_usb_storage_gadget:s0
+/sys/devices/virtual/host_notify/usb_otg(/.*)?               u:object_r:sysfs_usb_otg:s0
diff --git a/sepolicy/common/init.te b/sepolicy/common/init.te
index 37b8139..d3d7732 100644
--- a/sepolicy/common/init.te
+++ b/sepolicy/common/init.te
@@ -1,22 +1,34 @@
+selinux_check_context(init)
+
+allow init hal_sensors_hwservice:hwservice_manager { add find };
+
+allow init hidl_base_hwservice:hwservice_manager add;
+
+allow init iio_device:chr_file ioctl;
+
+allow init sysfs_graphics:file r_file_perms;
+
 allow init {
     sysfs_iio
     sysfs_sec_tsp
     sysfs_sensors
 }:lnk_file read;
 
-allow init sysfs_graphics:file r_file_perms;
-
 allow init {
     sysfs_input
     sysfs_sensors
 }:file rw_file_perms;
 
 allow init {
+    sysfs_audio
     sysfs_batteryinfo
     sysfs_bluetooth_writable
+    sysfs_camera
     sysfs_graphics
+    sysfs_hal_pwr
     sysfs_iio
     sysfs_input
+    sysfs_kgsl
     sysfs_leds
     sysfs_mdnie
     sysfs_msmuart_file
@@ -32,10 +44,18 @@ allow init {
     sysfs_sec_tsp
     sysfs_sensors
     sysfs_socinfo
+    sysfs_usb_otg
+    sysfs_usb_storage_gadget
+    sysfs_wifi_writeable
 }:file setattr;
 
 allow init {
+    sysfs_cpu_boost
+    sysfs_devices_system_cpu
+    sysfs_lowmemorykiller
     sysfs_mmc_host
     sysfs_msm_perf
+    sysfs_msm_power
     sysfs_sec_key
+    sysfs_thermal
 }:file w_file_perms;
-- 
2.17.1

